cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/Transform.java

```diff
@@ -61,7 +62,29 @@ public void destroy() {
     //no-op
   }
 
+  /**
+   * Prepare the Batch run. Used to configure the job before starting the run.
+   *
+   * @param context transform prepare context
+   * @throws Exception if there's an error during this method invocation
+   */
+  public void prepareRun(TransformPrepareContext context) throws Exception {
+    // no-op
+  }
+
+  /**
+   * Invoked after the Batch run finishes. Used to perform any end of the run logic.
+   *
+   * @param succeeded defines the result of batch execution: true if run succeeded, false otherwise
+   * @param context transform prepare context
+   */
+  public void onRunFinish(boolean succeeded, TransformPrepareContext context) {
```

Comment:
`TransformPrepareContext` needs to be renamed, since its naming doesn't fit the usage here (in `onRunFinish` method).